connectors/sources/sharepoint_server.py

```diff
@@ -480,16 +480,16 @@ def get_default_configuration(cls):
             },
             "ssl_enabled": {
                 "display": "toggle",
-                "label": "Enable SSL",
+                "label": "Enable SSL certificate verification",
                 "order": 5,
                 "type": "bool",
-                "value": False,
+                "value": True,
             },
             "ssl_ca": {
-                "depends_on": [{"field": "ssl_enabled", "value": True}],
```

Comment:
Why remove depends_on?

Reply:
Because right now, with that `depends_on`, if you want to enable SSL then you must also provide your own cert (see screenshot below), which to me does not make sense. A quite common scenario would be (1) enable SSL but (2) do not provide your own cert.

I'm trying in this PR to decouple those two things, since I don't understand why this connector was written in such a way that providing your own cert is coupled to also having SSL enabled.

![image](https://github.com/elastic/connectors/assets/10353074/107b702b-9a76-4c5c-b805-5582c6b03874)